packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -520,63 +533,60 @@ export function TransferPanel() {
     }
 
     try {
-      const sourceChainId = latestNetworks.current.sourceChain.id
-      const waitForInput = isDeposit
-        ? openUSDCDepositConfirmationDialog()
-        : openUSDCWithdrawalConfirmationDialog()
-      const [confirmed, primaryButtonClicked] = await waitForInput()
+      const { sourceChainProvider, destinationChainProvider } = networks
 
-      if (!confirmed) {
-        return
-      }
-      if (isDeposit && primaryButtonClicked === 'bridged') {
-        // User has selected normal bridge (USDC.e)
-        depositToken()
-        return
+      const sourceChainId = await getChainIdFromProvider(sourceChainProvider)
```

Comment:
you can just use `sourceChain.id`, it's returned from `useNetworks`

Reply:
Updated